tool/gravity/cli/checks.go

```diff
@@ -57,15 +78,91 @@ func checkManifest(env *localenv.LocalEnvironment, manifestPath, profileName str
 		fixableErr = trace.BadParameter(fmt.Sprintf("The following checks failed, provide --autofix flag to let gravity to autofix them:\n%v",
 			checks.FormatFailedChecks(result.Fixable)))
 	}
-
 	return trace.NewAggregate(failedErr, fixableErr)
 }
 
+func checkUpgrade(ctx context.Context, env *localenv.LocalEnvironment, manifestPath, imagePath string) error {
+	tarballEnv, err := localenv.NewTarballEnvironment(localenv.TarballEnvironmentArgs{
+		StateDir: imagePath,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	operator, err := env.SiteOperator()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	apps, err := env.SiteApps()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	packages, err := env.ClusterPackages()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Need to upload gravity package from the upgrade image, otherwise
+	// RPC agents may fail to deploy because they will be looking for
+	// this gravity package in the cluster's package service.
+	err = uploadGravity(ctx, env, tarballEnv.Packages, packages)
```

Comment:
Why not use the manifest and deal with the specific gravity package version? You won't need `FindPackageByName` in this case.